### PR TITLE
Make it idempotent to capturePromise

### PR DESCRIPTION
**Pull request by fossamagna**
*Issue #, if available:*
#398

*Description of changes:*
Make it idempotent to capturePromise.
Add uncapturePromise function in order to cancel patch in tests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/test/integration/segment_maintained_across_shared_promise.test.js [hunk 1]
```diff
@@ -70,3 +70,6 @@ function sendRequest(address, cb) {
     .on('error', cb)
     .end();
 }
+
+// Cancel the patch because it doesn't affect other tests
+require('../../lib/patchers/promise_p').uncapturePromise();
```
Comment (willarmiros) [inline packages/core/test/integration/segment_maintained_across_shared_promise.test.js:75]: So this is effectively cleanup for the integration test? If so, can we add it after the assertion so we're confident it's executed after the test will execute? Or perhaps wrap things in a `try..finally` block? Please inform me if I'm misunderstanding the purpose here.

**Review by willarmiros — COMMENTED** [review]
Thanks for the contribution! Generally lgtm, just one comment/question.

File: packages/core/test/integration/segment_maintained_across_shared_promise.test.js (hunk 1 above)
Comment (fossamagna) [reply packages/core/test/integration/segment_maintained_across_shared_promise.test.js]: @willarmiros Thank you for your review. I moved cleanup for integration test after assertion, and change setup and cleanup process in `promise_p.test.js` too.

**Review by willarmiros — APPROVED** [review]
Thanks!

**Pull request merged**